### PR TITLE
Add tooltip `group` param

### PR DESCRIPTION
**Pull request by abdel-17**
This new parameter allows tooltip to be grouped together, coupling their `open` states so that no more than 1 tooltip can be open in a single group.

This pr includes changes to the `createTooltip` function and its types, and some unit tests to validate the functionality.

### CODE REVIEW

**Comment by abdel-17** [comment]
Okay, so somehow the previous slider docs pr got merged into this one and I don't know how to fix it. Sorry for that.

File: src/lib/builders/tooltip/create.ts [hunk 1]
```diff
@@ -40,11 +40,15 @@ const defaults = {
...
 	closeOnEscape: true,
 	disableHoverableContent: false,
+	group: undefined,
 } satisfies CreateTooltipProps;
 
 type TooltipParts = 'trigger' | 'content' | 'arrow';
 const { name } = createElHelpers<TooltipParts>('tooltip');
 
+// Store a global map to get the currently open tooltip.
+const _groupMap = new Map<string | boolean, Writable<boolean>>();
```
Comment (TGlide) [inline src/lib/builders/tooltip/create.ts:50]: Probably should be `string | true`, as we don't want `false` to be a valid group option

File: src/lib/builders/tooltip/create.ts [hunk 2]
```diff
@@ -240,6 +245,21 @@ export function createTooltip(props?: CreateTooltipProps) {
...
+			if (groupValue === undefined || groupValue === false) {
+				return;
+			}
+
+			// Close the currently open tooltip in the same group
+			// and set this tooltip as the open one.
+			const currentOpen = _groupMap.get(groupValue);
+			currentOpen?.set(false);
+			_groupMap.set(groupValue, open);
+		}
```
Comment (TGlide) [inline src/lib/builders/tooltip/create.ts:260]:
```suggestion
		if (!$open) return
		const groupValue = get(group);
		if (groupValue === undefined || groupValue === false) {
			return;
		}

		// Close the currently open tooltip in the same group
		// and set this tooltip as the open one.
		const currentOpen = _groupMap.get(groupValue);
		currentOpen?.set(false);
		_groupMap.set(groupValue, open);
	
```

File: src/lib/builders/tooltip/create.ts (hunk 1 above)
Comment (TGlide) [reply src/lib/builders/tooltip/create.ts:50]: Also, why the underline?

**Review by TGlide — COMMENTED** [review]
Thank you for the PR! Please don't forrget to add documentation for the group feature :)

File: src/lib/builders/tooltip/create.ts (hunk 1 above)
Comment (abdel-17) [reply src/lib/builders/tooltip/create.ts:50]: I did handle the `false` case below, where it acts similar to `undefined`. I think it's a valid choice as both as "falsy", but I can remove it if you like.

The underline, eh, idk? It's internal global state, I just did it out of habit. I'll remove it.

**Comment by abdel-17** [comment]
> Thank you for the PR! Please don't forrget to add documentation for the group feature :)

Hey! I wanted to add documentation, but couldn't figure out a way to clearly explain what `group` does. Can you suggest an explanation to add to the JSDoc? I'll add it to the documentation site.

**Comment by TGlide** [comment]
> > Thank you for the PR! Please don't forrget to add documentation for the group feature :)
> 
> Hey! I wanted to add documentation, but couldn't figure out a way to clearly explain what `group` does. Can you suggest an explanation to add to the JSDoc? I'll add it to the documentation site.

Maybe:

> When `true`, whenever you open a tooltip, all other tooltips with `group` also set to `true` will close. If you pass in a `string` instead, only tooltips with the same `group` value will be closed.

**Comment by abdel-17** [comment]
> > > Thank you for the PR! Please don't forrget to add documentation for the group feature :)
> > 
> > 
> > Hey! I wanted to add documentation, but couldn't figure out a way to clearly explain what `group` does. Can you suggest an explanation to add to the JSDoc? I'll add it to the documentation site.
> 
> Maybe:
> 
> > When `true`, whenever you open a tooltip, all other tooltips with `group` also set to `true` will close. If you pass in a `string` instead, only tooltips with the same `group` value will be closed.

Hey! I add `group` to the documentation. I wanted to add an example, but the Tooltip didn't have any and I'm not familiar with how to add one, so I'll leave this one up to you if you want to add one.

**Pull request merged**